### DISCUSSION TITLE
Trivy config scan on helm charts > authorization-policy ksv116 false positive

### DISCUSSION THREAD

**Discussion by Mo0rBy** [aquasecurity/trivy#5132] [Bugs]
### Description

When performing a `trivy config` scan on my packaged helm chart, I am getting an alert for ksv116:

authorizationpolicy (my-policy-name) in default namespace should set spec.securityContext.runAsGroup, spec.securityContext.supplementalGroups[*] and spec.securityContext.fsGroup to integer greater than 0

When looking at the GO packages for `security.istio.io/v1beta1` authorization-policy object, it doesn't look like you can add a securityContext to it, which is what I suspected.

### Desired Behavior

An istio authorization-policy object **_does not_** trigger an alert for ksv116

### Actual Behavior

An istio authorization-policy object **_does_** trigger an alert for ksv116

### Reproduction Steps

```bash
1. Have a helm chart (I suppose you could also just have a yaml file) that includes an istio authorization-policy object
2. Package the helm chart
3. Run `trivy config <helm-chart-package>`
```

### Target

Kubernetes

### Scanner

Misconfiguration

### Output Format

Table

### Mode

Standalone

### Debug Output

```bash
No need for this
```

### Operating System

macOS Ventura 13.5.1

### Version

```bash
Version: 0.44.1
Vulnerability DB:
  Version: 2
  UpdatedAt: 2023-09-04 06:15:04.597487258 +0000 UTC
  NextUpdate: 2023-09-04 12:15:04.597486858 +0000 UTC
  DownloadedAt: 2023-09-04 09:25:11.024977 +0000 UTC
Java DB:
  Version: 1
  UpdatedAt: 2023-08-24 00:51:26.692869902 +0000 UTC
  NextUpdate: 2023-08-27 00:51:26.692869302 +0000 UTC
  DownloadedAt: 2023-08-24 13:07:25.423683 +0000 UTC
Policy Bundle:
  Digest: sha256:fd5f1ce3d8efb1fe158cb41f9adb9d7c7cc5c4c863b261053c962e6d950350b3
  DownloadedAt: 2023-09-06 15:49:34.798831 +0000 UTC
```

### Checklist

- [ ] Run `trivy image --reset`
- [ ] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by Mo0rBy** [comment]
This is very similar to #4922 but this is for istio objects, not native k8s objects

**Comment by simar7** [comment]
cc @chen-keinan

**Comment by Mo0rBy** [comment]
In the mean time, is there a way to ignore this alert for this specific yaml file and not globally across all yaml files in the chart?
I'm trying to find a way in the docs. I don't really want to add an inline ignore comment in the yaml's (if that's possible, I see you can do that for terraform code). I'd prefer to define like:

For X filename or filepath, ignore the alert for Y rule (in this case, on the authorization-policy.yaml, ignore ksv116)

It looks like creating a custom policy could be what I need, but you can only add new custom rules, not custom fine-tuned ignores (https://aquasecurity.github.io/trivy/v0.45/docs/scanner/misconfiguration/custom/)

**Comment by Mo0rBy** [comment]
This rule alert is also pooping up for the istio request-authentication object. Same rule, ksv116, and it is not possible to add a securityContext to the request-authentication object.

**Comment by Mo0rBy** [comment]
I've made a new post with a much better summary of all the false positives I am seeing and the digging I've done to prove why they are false positives. The new post is #5174. I will close this discussion now as this new one is more concise and detailed.
